src/core/unit-varlink.c

```diff
@@ -311,18 +314,114 @@ static int list_unit_one(sd_varlink *link, Unit *unit, bool more) {
         return sd_varlink_reply(link, v);
 }
 
+static int lookup_unit_by_name(sd_varlink *link, Manager *manager, const char *name, Unit **ret_unit) {
+        assert(link);
+        assert(manager);
+        assert(ret_unit);
+
+        if (!name)
+                return -EINVAL;
+        if (!unit_name_is_valid(name, UNIT_NAME_PLAIN|UNIT_NAME_INSTANCE))
+                return -EINVAL;
+
+        Unit *u = manager_get_unit(manager, name);
+        if (!u)
+                return -ESRCH;
```

Comment:
With the above two suggestions this function doesn't do a whole lot anymore so maybe just inline it